exporters/otlp/include/opentelemetry/exporters/otlp/otlp_grpc_exporter_options.h

```diff
@@ -6,13 +6,44 @@
 #include "opentelemetry/exporters/otlp/otlp_environment.h"
 
 #include <memory>
+// TODO: These requires C++17, is that OK? It seems like we need to support C++0x?
+#include <optional>
+#include <variant>
 
 OPENTELEMETRY_BEGIN_NAMESPACE
 namespace exporter
 {
 namespace otlp
 {
 
+// This type allows the variants to be default-constructed to neither type.
+// This avoids needing deeper template if std::optional<std::variant<>>.
+struct Unset {};
+
+// TODO: Check if this strong-typing is ok. It seems nice, and allows variant typing.
+struct FilePath {
+  std::string value;
```

Comment:
I think we'd better use `nostd::string_view` here.

Reply:
Just note that the existing options in OtlpGrpcExporterOptions all use std::string, hence why I used that. If this should be different, so be it - just checking before changing it.